java/src/jmri/jmrit/operations/trains/tools/ExportTimetable.java

```diff
@@ -0,0 +1,294 @@
+ *            Defaults:  "New Schedule", "Today", 0, 24
+ *            Occurs: Follows last station, occurs 1 to n times.
+ *
+ * "Train", "train name", "train description", type, defaultSpeed, starttime, throttle, notes
+ *            Defaults:  "NT", "New Train", 0, 1, 0, 0, ""
+ *            Occurs:  Follows parent schedule, occurs 1 to n times.
+ *            Note1:  The type is the relative number of the train type listed above starting with 1 for the first train type.
+ *            Note2:  The start time is an integer between 0 and 1439, subject to the schedule start time and duration.
+ *
+ * "Stop", station, duration, nextSpeed, stagingTrack, notes
+ *            Defaults:  0, 0, 0, 0, ""
+ *            Required: station number.
+ *            Occurs:  Follows parent train in the proper sequence.  Occurs 1 to n times.
+ *            Notes:  The station is the relative number of the station listed above starting with 1 for the first station.
+ *                    If more that one segment is used, the station number is cumulative.
+ *
+ * Except for Stops, each record can have one of three actions:
+ *    1) If no name is supplied, a default object will be created.
+ *    2) If the name matches an existing name, the existing object will be used.
+ *    3) A new object will be created with the supplied name.  The remaining fields, if any, will replace the default values.
+ *
+ * Minimal file using defaults except for station names and distances:
+ * "Layout"
+ * "Segment"
+ * "Station", "Station 1", 0.0
+ * "Station", "Station 2", 25.0
+ * "Schedule"
+ * "Train"
+ * "Stop", 1
+ * "Stop", 2
+ * </pre>
+ */
+public class ExportTimetable extends XmlFile {
+
+    static final String ESC = "\""; // escape character NOI18N
+    private String del = ","; // delimiter
+
+    public ExportTimetable() {
+
+    }
+
+    public void setDeliminter(String delimiter) {
+        del = delimiter;
+    }
+
+    public void writeOperationsTimetableFile() {
+        makeBackupFile(defaultOperationsFilename());
+        try {
+            if (!checkFile(defaultOperationsFilename())) {
+                // The file does not exist, create it before writing
+                java.io.File file = new java.io.File(defaultOperationsFilename());
+                java.io.File parentDir = file.getParentFile();
+                if (!parentDir.exists()) {
+                    if (!parentDir.mkdir()) {
+                        log.error("Directory wasn't created");
+                    }
+                }
+                if (file.createNewFile()) {
+                    log.debug("File created");
+                }
+            }
+            writeFile(defaultOperationsFilename());
+        } catch (Exception e) {
+            log.error("Exception while writing the new CSV operations file, may not be complete: " + e);
+        }
+    }
+
+    @SuppressFBWarnings(value = "OS_OPEN_STREAM_EXCEPTION_PATH", justification="Someone smarter than me can fix")
```

Comment:
This isn’t a valid reason for ignoring a warning.  You need to use a try with resources block